app/src/pages/playground/ModelComboBox.tsx

```diff
@@ -0,0 +1,92 @@
+import React, { useMemo, useState } from "react";
+import { graphql, useFragment } from "react-relay";
+
+import { ComboBox, ComboBoxItem, ComboBoxProps } from "@phoenix/components";
+
+import { ModelComboBoxFragment$key } from "./__generated__/ModelComboBoxFragment.graphql";
+
+type ModelComboBoxProps = {
+  query: ModelComboBoxFragment$key;
+  onChange: (model: string) => void;
+  provider: ModelProvider;
+  modelName: string | null;
+  container?: HTMLElement;
+} & Omit<
+  ComboBoxProps<{ name: string }>,
+  "children" | "onSelectionChange" | "defaultSelectedKey"
+>;
+
+export function ModelComboBox({
+  query,
+  onChange,
+  container,
+  ...props
+}: ModelComboBoxProps) {
+  const data = useFragment<ModelComboBoxFragment$key>(
+    graphql`
+      fragment ModelComboBoxFragment on Query
+      @argumentDefinitions(
+        providerKey: { type: "GenerativeProviderKey!", defaultValue: OPENAI }
+      ) {
+        models(input: { providerKey: $providerKey }) {
+          name
+        }
+      }
+    `,
+    query
+  );
+
+  const [modelInput, setModelInput] = useState("");
+
+  const items = useMemo(() => {
+    const items = data.models.map((model) => ({
+      name: model.name,
+      id: model.name,
+    }));
+
+    if (
+      modelInput !== "" &&
+      !items.some(
+        (model) =>
+          model.name.toLocaleLowerCase() === modelInput.toLocaleLowerCase()
+      )
+    ) {
+      items.push({ name: modelInput, id: modelInput });
```

Comment:
Can we move this to the top of the list, tweaking the label like `New model: ${modelInput}`?

Makes it a little more clear to me that you are "adding" a model to the list, rather than just filtering the current set of models

Reply:
I don't think we need an extra label `New model:` because i think that should be pretty clear to the user that it's selectable when they interact with the menu, but i can move it to the top